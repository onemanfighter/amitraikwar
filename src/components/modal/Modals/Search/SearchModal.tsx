import {
  Box,
  Input,
  InputGroup,
  InputLeftElement,
  useColorMode,
} from '@chakra-ui/react';
import { SearchIcon } from '@assets';
import Fuse from 'fuse.js';
import { useEffect, useState } from 'react';
import { fuse } from '@providers';

const SearchModal = () => {
  const { colorMode } = useColorMode();
  const [searchText, setSearchText] = useState('');
  const [searchResults, setSearchResults] = useState<any[]>([]);

  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchText(event.target.value);
  };

  useEffect(() => {
    const searchResults = fuse.search(searchText);
    setSearchResults(searchResults);
  }, [searchText]);

  return (
    <Box padding={1} top={100} zIndex={1000}>
      <InputGroup>
        <InputLeftElement pointerEvents="none">
          <SearchIcon />
        </InputLeftElement>
        <Input
          type="search"
          placeholder="Search anything related to me"
          variant={'ghost'}
          value={searchText}
          onChange={handleSearch}
        />
      </InputGroup>

      <Box
        maxHeight={400}
        overflowY="auto"
        border="1px"
        borderRadius={4}
        boxShadow="md"
      >
        {searchResults.map((result, index) => (
          <Box
            key={index}
            padding={2}
            _hover={{
              bgGradient:
                colorMode === 'light'
                  ? 'linear(to-r, green.200, blue.500)'
                  : 'linear(to-r, green.800, blue.900)',
              cursor: 'pointer',
            }}
            _active={{
              bgGradient:
                colorMode === 'light'
                  ? 'linear(to-r, green.200, blue.500)'
                  : 'linear(to-r, green.700, blue.800)',
            }}
            onClick={() => {
              console.log(result.item);
            }}
          >
            {result.item.title}
          </Box>
        ))}
      </Box>
    </Box>
  );
};

export default SearchModal;
