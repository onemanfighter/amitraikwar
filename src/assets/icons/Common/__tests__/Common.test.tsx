import { render } from '@testing-library/react';
import { SearchIcon } from '../Search';
import { ThreeDIcon } from '../3D';
import { TwoDIcon } from '../2D';

describe('Common Icon', () => {
  it(' `SearchIcon` renders correctly', () => {
    const { container } = render(<SearchIcon />);

    expect(container).toMatchSnapshot();
  });

  it(' `ThreeDIcon` renders correctly', () => {
    const { container } = render(<ThreeDIcon />);

    expect(container).toMatchSnapshot();
  });

  it(' `TwoDIcon` renders correctly', () => {
    const { container } = render(<TwoDIcon />);

    expect(container).toMatchSnapshot();
  });
});
