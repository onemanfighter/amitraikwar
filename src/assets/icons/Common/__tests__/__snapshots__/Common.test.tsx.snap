// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Icon  \`SearchIcon\` renders correctly 1`] = `
<div>
  <svg
    class="w-6 h-6"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`Common Icon  \`ThreeDIcon\` renders correctly 1`] = `
<div>
  <svg
    class="w-6 h-6"
    fill="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12.378 1.602a.75.75 0 0 0-.756 0L3 6.632l9 5.25 9-5.25-8.622-5.03ZM21.75 7.93l-9 5.25v9l8.628-5.032a.75.75 0 0 0 .372-.648V7.93ZM11.25 22.18v-9l-9-5.25v8.57a.75.75 0 0 0 .372.648l8.628 5.033Z"
    />
  </svg>
</div>
`;

exports[`Common Icon  \`TwoDIcon\` renders correctly 1`] = `
<div>
  <svg
    class="w-6 h-6"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m21 7.5-9-5.25L3 7.5m18 0-9 5.25m9-5.25v9l-9 5.25M3 7.5l9 5.25M3 7.5v9l9 5.25m0-9v9"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;
